include/libdnf5/comps/comps_sack.hpp

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef LIBDNF5_COMPS_COMPS_SACK_HPP
+#define LIBDNF5_COMPS_COMPS_SACK_HPP
+
+#include "libdnf5/base/base_weak.hpp"
+#include "libdnf5/comps/environment/query.hpp"
+#include "libdnf5/comps/group/query.hpp"
+#include "libdnf5/defs.h"
+
+#include <set>
+#include <string>
+
+namespace libdnf5::comps {
+
+
+class CompsSack;
+using CompsSackWeakPtr = WeakPtr<CompsSack, false>;
+
+class LIBDNF_API CompsSack {
+public:
+    explicit CompsSack(const libdnf5::BaseWeakPtr & base);
+    explicit CompsSack(libdnf5::Base & base);
+    ~CompsSack();
+
+    /// Create WeakPtr to CompsSack
+    CompsSackWeakPtr get_weak_ptr();
+
+    /// @return The `Base` object to which this object belongs.
+    /// @since 5.2
+    libdnf5::BaseWeakPtr get_base() const;
+
+    /// Loads excluded comps from the configuration.
+    /// Uses the `disable_excludes`, `excludegroups` and `excludeenvironments` configuration options for calculation.
+    /// @since 5.2
+    void load_config_excludes();
```

Comment:
I suppose we might also have `load_config_includes` or combine these to `load_config_excludes_includes` like package_sack in the future.

Reply:
I didn't add includes for groups, because I haven't found a use case for it and I didn't want to add unnecessary config options. But maybe I could rename the method to `load_config_excludes_includes`, so that it's general just in case the options will be added in the future... What do you think?

Reply:
I think it's fine as-is, calling it `load_config_excludes_includes` now might be misleading.